functions/Environment/fn_campShowOSD.sqf

```diff
@@ -0,0 +1,131 @@
+/*
+ * client side function that displays message of current location with time and
+ * possibly some other related info.
+ *
+ * Parameter(s):
+ * _this select 0: array (optional)    - position (default: player's position)
+ * _this select 1: array (optional)    - date in format [_year,_month,_day,_hour,_min] (default: current date)
+ * _this select 2: string (optional)   - optional message
+ *
+ * A3 fonts used:
+ *
+ * PuristaLight
+ * PuristaMedium
+ * PuristaSemiBold
+ * PuristaBold
+ *
+ * Following example displays current player position and time:
+ *
+ * [] call A3MT_fnc_campShowOSD;
+ *
+ * Note that function requires to be run in scheduled environment with call,
+ * or use spawn in unscheduled.
+ *
+*/
+
+//get current location text
+private _get_location = {
+    params ["_position"];
+    private["_locations","_loc","_nearest_location"];
+
+    _locations = nearestLocations [_position, ["NameCity", "NameCityCapital", "NameLocal", "NameMarine", "NameVillage"], 500];
+
+    // filter-out locations without names
+    {
+        if (text _x == "") then
+        {
+            _locations set [_forEachIndex, objNull];
+        };
+    } forEach _locations;
```

Comment:
This seems like a very weird way of filtering out the location list, can't you just iterate over what `nearesLocations` returns and if it has a name, add it to a list of locations?